### PR TITLE
Make activations strongly typed as an Enum

### PR DESCRIPTION
**Pull request by danieldk**
## Description

Stop passing around activations as strings.

### Types of change

Maintenance.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by danieldk** [comment]
I was thinking about this more and I should probably change the feedforward layer to take a `Module` for the nonlinearity and do the conversion earlier. I’ll change the PR to do this.

File: curated_transformers/layers/activations.py [hunk 1]
```diff
@@ -1,11 +1,96 @@
...
+
+    #: Gaussian Error Linear Unit (`Hendrycks et al., 2016`_) approximation.
+    GELUNew = "gelu_new"
+
+    #: Sigmoid Linear Unit (`Hendrycks et al., 2016`_).
+    SiLU = "silu"
+
+    @property
+    def module(self) -> Type[torch.nn.Module]:
+        """Get the PyTorch module for the activation function."""
```
Comment (shadeMe) [inline curated_transformers/layers/activations.py:76]:
```suggestion
        """
        Get the PyTorch module for the activation function.
        """
```

File: curated_transformers/layers/feedforward.py [hunk 2]
```diff
@@ -4,17 +4,17 @@
 from torch import Tensor
 from torch.nn import Linear, Module
 
-from .activations import GeluFast, GeluNew
+from .activations import Activation, GELUFast, GELUNew
 
 _ACTIVATIONS: Dict[str, Type[Module]] = {
```
Comment (shadeMe) [inline curated_transformers/layers/feedforward.py:9]: Unused.

File: curated_transformers/layers/feedforward.py [hunk 3]
```diff
@@ -55,10 +55,8 @@ def __init__(
         """
         Construct a point-wise feed-forward layer module.
```
Comment (shadeMe) [inline curated_transformers/layers/feedforward.py:56]:
```suggestion
        Construct a pointwise feed-forward layer module.
```

File: curated_transformers/models/config.py [hunk 4]
```diff
@@ -1,6 +1,10 @@
 from dataclasses import dataclass
 from typing import Optional
 
+from torch.nn.modules import activation
+
+from curated_transformers.layers.activations import Activation
```
Comment (shadeMe) [inline curated_transformers/models/config.py:6]: Relative import.

**Comment by shadeMe** [comment]
Sorry, didn't notice that you had reverted the PR to draft before publishing the review.

File: curated_transformers/layers/feedforward.py (hunk 2 above)
Comment (danieldk) [reply curated_transformers/layers/feedforward.py]: Removed.

File: curated_transformers/models/config.py (hunk 4 above)
Comment (danieldk) [reply curated_transformers/models/config.py]: Fix this and some other absolute imports.

**Review by shadeMe — APPROVED** [review]

**Pull request merged**